staging/src/k8s.io/kubectl/pkg/cmd/create/create_pdb.go

```diff
@@ -60,54 +95,160 @@ func NewCmdCreatePodDisruptionBudget(f cmdutil.Factory, ioStreams genericcliopti
+	cmd.Flags().StringVar(&o.MinAvailable, "min-available", o.MinAvailable, i18n.T("The minimum number or percentage of available pods this budget requires."))
+	cmd.Flags().StringVar(&o.MaxUnavailable, "max-unavailable", o.MaxUnavailable, i18n.T("The maximum number or percentage of unavailable pods this budget requires."))
+	cmd.Flags().StringVar(&o.Selector, "selector", o.Selector, i18n.T("A label selector to use for this budget. Only equality-based selector requirements are supported."))
+	cmdutil.AddFieldManagerFlagVar(cmd, &o.FieldManager, "kubectl-create")
 	return cmd
 }
 
 // Complete completes all the required options
 func (o *PodDisruptionBudgetOpts) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	name, err := NameFromCommandArgs(cmd, args)
+	var err error
+	o.Name, err = NameFromCommandArgs(cmd, args)
 	if err != nil {
 		return err
 	}
 
-	var generator generate.StructuredGenerator
-	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
-	case generateversioned.PodDisruptionBudgetV1GeneratorName:
-		generator = &generateversioned.PodDisruptionBudgetV1Generator{
-			Name:         name,
-			MinAvailable: cmdutil.GetFlagString(cmd, "min-available"),
-			Selector:     cmdutil.GetFlagString(cmd, "selector"),
-		}
-	case generateversioned.PodDisruptionBudgetV2GeneratorName:
-		generator = &generateversioned.PodDisruptionBudgetV2Generator{
-			Name:           name,
-			MinAvailable:   cmdutil.GetFlagString(cmd, "min-available"),
-			MaxUnavailable: cmdutil.GetFlagString(cmd, "max-unavailable"),
-			Selector:       cmdutil.GetFlagString(cmd, "selector"),
-		}
-	default:
-		return errUnsupportedGenerator(cmd, generatorName)
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.Client, err = policyclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
 	}
 
-	return o.CreateSubcommandOptions.Complete(f, cmd, args, generator)
+	o.CreateAnnotation = cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag)
+
+	o.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	o.DryRunVerifier = resourcecli.NewDryRunVerifier(dynamicClient, discoveryClient)
+
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+// Validate checks to the PodDisruptionBudgetOpts to see if there is sufficient information run the command
+func (o *PodDisruptionBudgetOpts) Validate() error {
```

Comment:
Here, doing some tests:

When I create a PDB with invalid values, like "--min-available=10gb" and a client dryrun it generates the yaml, but with server dry-run not (because the API Server validates the input with the regex `[0-9]+%`

I think, even if this is not the focus of this PR, that adding this validation inside the Validate() (and also unit tests for this) would add some more value to the create command, so much more as a recommendation then a request for change, what about adding this validation for --min-available and --max-available here in validate?

Reply:
I've created a validation for the above at the end of `Validate()` which looks like this:
```go
re := regexp.MustCompile(`^[0-9]+%?$`)

switch {
case len(o.MinAvailable) > 0 && !re.MatchString(o.MinAvailable):
    return fmt.Errorf("invalid format specified for min-avaiable")
case len(o.MaxUnavailable) > 0 && !re.MatchString(o.MaxUnavailable):
    return fmt.Errorf("invalid format specified for max-unavailable")
}
```
But I'm having a little problem writing the unit test for the same. Should I rewrite them by leveraging a fake HTTP server as was done before for testings? Please suggest if the test can be written in this table-driven testing format.

Reply:
No need to make fake HTTP.

What you can do here is like https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go#L28

Create another set of tests, but only for the validation function, then create a set of scenarios with the correct and the incorrect values (like minAvailable > 0 but not matching the regex, then matching the regex, then both matching the regex which should raise an error because of the existing of both arguments)

Reply:
Done!